File: CS/ButtonEditWithPadding/PaddingButtonEditPainter.cs
using System.Drawing;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace ButtonEditWithPadding
{
    class PaddingButtonEditPainter : ButtonEditPainter
    {
        public PaddingButtonEditPainter()
        {
        }

        protected override void DrawButton(ButtonEditViewInfo viewInfo, EditorButtonObjectInfoArgs info)
        {
            if ( viewInfo is PaddingButtonEditViewInfo )
            {
                Brush b = viewInfo.PaintAppearance.GetBackBrush(info.Cache);
                info.Graphics.FillRectangle(b, info.Bounds);

                info.Bounds = ((PaddingButtonEditViewInfo)viewInfo).AdjustBoundsWithMargins(info.Bounds);
                base.DrawButton(viewInfo, info);
                info.Bounds = ((PaddingButtonEditViewInfo)viewInfo).RevertMarginedBounds(info.Bounds);
            }
            else
                base.DrawButton(viewInfo, info);
        }
    }
}

File: VB/ButtonEditWithPadding/PaddingButtonEditPainter.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace ButtonEditWithPadding
	Friend Class PaddingButtonEditPainter
		Inherits ButtonEditPainter
		Public Sub New()
		End Sub

		Protected Overrides Sub DrawButton(ByVal viewInfo As ButtonEditViewInfo, ByVal info As EditorButtonObjectInfoArgs)
			If TypeOf viewInfo Is PaddingButtonEditViewInfo Then
				Dim b As Brush = viewInfo.PaintAppearance.GetBackBrush(info.Cache)
				info.Graphics.FillRectangle(b, info.Bounds)

				info.Bounds = (CType(viewInfo, PaddingButtonEditViewInfo)).AdjustBoundsWithMargins(info.Bounds)
				MyBase.DrawButton(viewInfo, info)
				info.Bounds = (CType(viewInfo, PaddingButtonEditViewInfo)).RevertMarginedBounds(info.Bounds)
			Else
				MyBase.DrawButton(viewInfo, info)
			End If
		End Sub
	End Class
End Namespace
